DeviceCode/Targets/Native/STM32F4/DeviceCode/Initialization/tinyhal.cpp

```diff
@@ -0,0 +1,752 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// This file is part of the Microsoft .NET Micro Framework Porting Kit Code Samples and is unsupported. 
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use these files except in compliance with the License.
+// You may obtain a copy of the License at:
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+// 
+#include <tinyhal.h>
+#include "stm32f4xx.h"
+
+#if defined(PLATFORM_ARM_OS_PORT)
+#include <cmsis_os.h>
+#include <cstdint>
+#endif
+
+// these define the region to zero initialize
+extern UINT32 Image$$ER_RAM_RW$$ZI$$Base;

```

Comment:
The Linker vairables Image$$... and Load$$... are specific to the ARM compiler and not generally useable on other tool chains. (i.e. gcc) furthermore we know there's a bug in the linker where it gets the Load$$ value stored in the literal pool wrong for this. Can we pull all of this this out to a toolset specific file at least?